### ISSUE TITLE
Challenge reward not correct after few times completing

### ISSUE THREAD

**Issue by dutchy1001** [rlf/uSkyBlock#362]
Challenge completing after the 4th time will not give the right amount repeat reward.

the challenge tested on was:
```
      caramba:
        description: "Il va faire tout vert."
        type: onPlayer
        requiredItems: '81:640'
        displayItem: '351:2'
        reward:
          text: "48 sables"
          items:  '12:48'
          currency: 150
          xp: 150
          commands:
          - 'console:effect {party} 1 60'
          - 'op:summon FireworksRocketEntity ~ ~ ~ {LifeTime:20}'
        repeatReward:
          text: "24 sables"
          items: '12:24'
          currency: 40
          xp: 40
```
From the 24 it should give normally as repeat reward  it only started to give 8 instead.

Usb v2.3-HF4c
All other plugins are up to date (but you knew that already)

**Comment by rlf** [comment]
My bet, the '12:24' is the problem, since yml has some difficulties with the : notation.
Not sure how to fix it though.

**Comment by CitoyenCharly** [comment]
The problem occurs when we farm more than 1 stack.

24 + 24 + 24 = 1 stack and 8 => Now the challenge bug and the next reward will be 8 until the next reboot.

32 + 32 = 1 stack => It's ok, the next reward will be 32.

But If I have 1 sand in my inventory :
1 + 32 + 32 = 1 stack and 1 => The next reward will be 1.

**Comment by rlf** [comment]
That doesn't make sense. Rewards are not (shouldn't be) dependent on the number of times the've been handed in - only the requiredItems does that.

I will take a look at the code, and try to re-create the "bug".

**Comment by rlf** [comment]
Confirmed - Debugging commencing.

**Comment by rlf** [comment]
Fixed and pushed - testabled in v2.3-HF4e

**Comment by CitoyenCharly** [comment]
It works! Thanks.
